snakemake_wrapper_utils/snakemake.py

```diff
@@ -28,20 +28,53 @@ def is_arg(arg, cmd):
 
 def get_format(path):
     from pathlib import Path
+
     """Get file format from extension, ignoring common compressions."""
     if not path:
         raise ValueError("Path cannot be empty")
-    exts = Path(path).suffixes
+    exts = [s.lower() for s in Path(path).suffixes]
     if not exts:
         raise ValueError("Path must have an extension")
-    if exts[-1] in (".gz", ".bgz", ".bz2"):
+    if exts[-1] in (".gz", ".bgz", ".bz2", ".xz"):
+        if len(exts) < 2:
+            raise ValueError(
+                "Compressed path must include a base extension before the compression suffix, e.g., '.vcf.gz'."
+            )
         ext = exts[-2]
     else:
         ext = exts[-1]
 
-    if ext in (".fq", ".fastq"):
+    ext = ext.lstrip(".")
+    if ext in ("fq", "fastq"):
         return "fastq"
-    elif ext in (".fa", ".fas", ".fna", ".fasta"):
+    elif ext in ("fa", "fas", "fna", "fasta"):
         return "fasta"
     else:
-        return ext.lstrip(".").lower()
+        return ext
+
+
+def move_files(snakemake, mapping, cmd="mv -v"):
+    """
+    Move file(s) produced by the tool to the named Snakemake outputs.
+
+    mapping must be a dict of {out_tag: source_path}.
+
+    Example:
+        mapping = {"tsv": "/tmp/tmp98723489/results/out.tsv"}
+
+    This moves /tmp/tmp98723489/results/out.tsv to snakemake.output["tsv"],
+    redirecting stdout/stderr to snakemake.log.
+    """
+
+    from snakemake.shell import shell
+
+    log = snakemake.log_fmt_shell(stdout=True, stderr=True, append=True)
+
+    for out_tag, tool_out_name in mapping.items():
+        out_name = snakemake.output.get(out_tag, "")
+        if out_name:
+            shell("{cmd} {tool_out_name:q} {out_name:q} {log}")
+        else:
+            raise KeyError(
+                f"The wrapper requires the named output: {out_tag}. Please provide this named output."
+            )
```

Comment:
Here's my suggestion to just generate a long string that can be stuck into a `shell()` statement in the main wrapper (I did not test this code :grimacing: ):

```suggestion
    for out_tag, tool_out_name in mapping.items():
        out_name = snakemake.output.get(out_tag, "")
        if not out_name:
            raise KeyError(
                f"The wrapper requires the named output: {out_tag}. Please provide this named output."
            )
    "; ".join([ f"{cmd} {move.tool_out_name:q} {move.out_tag:q}" for move in mapping ])
```